contracts/swaps/connectors/SwapsImplstETH_ETH.sol

```diff
@@ -152,7 +152,7 @@ contract SwapsImplstETH_ETH is State, ISwapsImpl {
         if (sourceTokenAddress == WETH) {
             IWeth(WETH).withdraw(minSourceTokenAmount);
             if (abi.decode(payload, (uint256)) > 0) {
```

Comment:
you are decoding `abi.decode` twice. you can extract this into a variable

Reply:
Yeah that should lower gas costs a tiny bit

Reply:
I will do it